ansible/roles/operator-pipeline/templates/openshift/tasks/github-add-comment.yml

```diff
@@ -92,82 +80,18 @@ spec:
-          commentParamValue = open(commentParamValue, "r").read()
-        # If a tag was specified, append it to the comment
-        if "$(params.COMMENT_TAG)":
-          commentParamValue += "<!-- {tag} -->".format(tag="$(params.COMMENT_TAG)")
-        data = {
-            "body": commentParamValue,
-        }
-        # This is for our fake github server
-        if "$(params.GITHUB_HOST_URL)".startswith("http://"):
-          conn = http.client.HTTPConnection("$(params.GITHUB_HOST_URL)".replace("http://", ""))
-        else:
-          conn = http.client.HTTPSConnection("$(params.GITHUB_HOST_URL)")
-        # If REPLACE is true, we need to search for comments first
-        matching_comment = ""
-        if "$(params.REPLACE)" == "true":
-          if not "$(params.COMMENT_TAG)":
-            print("REPLACE requested but no COMMENT_TAG specified")
-            sys.exit(1)
-          r = conn.request(
-            "GET",
-            api_url + "/comments",
-            headers={
-                "User-Agent": "TektonCD, the peaceful cat",
-                "Authorization": "Bearer " + os.environ["GITHUBTOKEN"],
-            })
-          resp = conn.getresponse()
-          if not str(resp.status).startswith("2"):
-              print("Error: %d" % (resp.status))
-              print(resp.read())
-              sys.exit(1)
-          print(resp.status)
-          comments = json.loads(resp.read())
-          print(comments)
-          # If more than one comment is found take the last one
-          matching_comment = [x for x in comments if '$(params.COMMENT_TAG)' in x['body']][-1:]
-          if matching_comment:
-              with open("$(results.OLD_COMMENT.path)", "w") as result_old:
-                result_old.write(str(matching_comment[0]))
-              matching_comment = matching_comment[0]['url']
-        if matching_comment:
-            method = "PATCH"
-            target_url = urllib.parse.urlparse(matching_comment).path
-        else:
-            method = "POST"
-            target_url = api_url + "/comments"
-        print("Sending this data to GitHub with {}: ".format(method))
-        print(data)
-        r = conn.request(
-            method,
-            target_url,
-            body=json.dumps(data),
-            headers={
-                "User-Agent": "TektonCD, the peaceful cat",
-                "Authorization": "Bearer " + os.environ["GITHUBTOKEN"],
-            })
-        resp = conn.getresponse()
-        if not str(resp.status).startswith("2"):
-            print("Error: %d" % (resp.status))
-            print(resp.read())
-        else:
-            with open("$(results.NEW_COMMENT.path)", "wb") as result_new:
-              result_new.write(resp.read())
-            print("a GitHub comment has been {} to $(params.REQUEST_URL)".format(
-              "updated" if matching_comment else "added"))
+        #! /usr/bin/env bash
+        set -xe -o pipefail
+
+        echo $params.COMMENT_OR_FILE > comment_file
```

Comment:
Since you always accept just a comment string can you rename the parameter to just `COMMENT` and also remove the workspace?

Reply:
Yes, everywhere.